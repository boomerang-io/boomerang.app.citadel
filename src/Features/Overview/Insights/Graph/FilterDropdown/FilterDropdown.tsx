import React from "react";
// @ts-expect-error ts-migrate(7016) FIXME: Could not find a declaration file for module '@boo... Remove this comment to see the full error message
import { Dropdown } from "@boomerang-io/carbon-addons-boomerang-react";

type Props = {
    label: string;
    options: any[];
    onChange: (...args: any[]) => any;
    selectedItem?: any;
};

export function FilterDropdown({ selectedItem, onChange, options, label }: Props) {
  return (
    // @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message
    <div style={{ width: "15rem", height: "3rem" }}>
      {/* @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message */}
      <Dropdown
        id="filter-dropdown"
        label={label}
        onChange={onChange}
        selectedItem={selectedItem}
        items={options}
        titleText={label}
      />
    </div>
  );
}

export default FilterDropdown;
