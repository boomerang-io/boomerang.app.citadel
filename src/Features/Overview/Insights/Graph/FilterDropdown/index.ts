// @ts-expect-error ts-migrate(6142) FIXME: Module './FilterDropdown' was resolved to 'C:/User... Remove this comment to see the full error message
export { default } from "./FilterDropdown"