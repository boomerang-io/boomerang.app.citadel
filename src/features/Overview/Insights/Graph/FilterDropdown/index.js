import React, { Component } from "react";
import PropTypes from "prop-types";
import { DropdownV2 } from "carbon-components-react";

export class FilterDropdown extends Component {
  static propTypes = {
    label: PropTypes.string.isRequired,
    options: PropTypes.array.isRequired,
    onChange: PropTypes.func.isRequired,
    selectedItem: PropTypes.object
  };

  render() {
    const { selectedItem, onChange, options, label } = this.props;
    return (
      <div style={{ width: "15rem", height: "3rem" }}>
        <DropdownV2 label={label} onChange={onChange} selectedItem={selectedItem} items={options} titleText={label} />
      </div>
    );
  }
}

export default FilterDropdown;
