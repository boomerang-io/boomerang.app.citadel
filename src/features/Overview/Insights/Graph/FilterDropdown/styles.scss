.b-graphs-dropdown-label{
  font-size: 1rem;
  font-weight: 500;
}
