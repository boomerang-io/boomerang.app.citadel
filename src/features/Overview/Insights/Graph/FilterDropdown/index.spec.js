import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import FilterDropdown from "./index";

const mockfn = jest.fn();

const label = "testLabel";
const options = [{text:"option1", value:"option1"}];
const onChange = mockfn;

describe("FilterDropdown --- Snapshot", () => {
  it("Capturing Snapshot of FilterDropdown", () => {
    const renderedValue = renderer
      .create(
        <FilterDropdown
          label={label}
          options={options}
          onChange={onChange}
        />
      )
      .toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});

describe("FilterDropdown --- Shallow render", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <FilterDropdown
        label={label}
        options={options}
        onChange={onChange}
      />
    );
  });

  it("Render the DUMB component", () => {
    expect(wrapper.length).toEqual(1);
  });
});
