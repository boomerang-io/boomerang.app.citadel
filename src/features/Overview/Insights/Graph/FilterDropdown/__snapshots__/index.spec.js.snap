// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterDropdown --- Snapshot Capturing Snapshot of FilterDropdown 1`] = `
<div
  style={
    Object {
      "height": "3rem",
      "width": "15rem",
    }
  }
>
  <div
    className="bx--dropdown__wrapper bx--list-box__wrapper"
  >
    <label
      className="bx--label"
      htmlFor="filter-dropdown"
    >
      testLabel
    </label>
    <div
      className="bx--dropdown bx--list-box"
      id="filter-dropdown"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="listbox"
      tabIndex="-1"
    >
      <div
        aria-controls={null}
        aria-disabled={false}
        aria-expanded={false}
        aria-haspopup={true}
        aria-label="Open menu"
        aria-owns={null}
        className="bx--list-box__field"
        data-toggle={true}
        onBlur={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex="0"
        type="button"
      >
        <span
          className="bx--list-box__label"
          htmlFor="downshift-0-input"
        >
          testLabel
        </span>
        <div
          className="bx--list-box__menu-icon"
        >
          <svg
            aria-label="Open menu"
            focusable="false"
            height={16}
            name="chevron--down"
            preserveAspectRatio="xMidYMid meet"
            role="img"
            style={
              Object {
                "willChange": "transform",
              }
            }
            viewBox="0 0 16 16"
            width={16}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 11L3 6l.7-.7L8 9.6l4.3-4.3.7.7z"
            />
            <title>
              Open menu
            </title>
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;
